### PR TITLE
Expose Batch Job Resource for Batch Deployments/Endpoints

### PR DESCRIPTION
**Pull request by TajiHarrisMicrosoft**
Expose batch job resources to sdk

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

Creating curated entity to allow sdk user to be exposed to Batch Job Resource class

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    :param system_data: System data associated with resource provider.
+    :type system_data: SystemData
+    """
+
+    def __init__(
+        self,
+        id: str = None,
+        name: str = None,
+        type: str = None,
+        properties: BatchJobProperty = None,
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py:36]: Sorry if I was not clear on this earlier. Can we promote fields of BatchJobProperty to top level as properties of BatchJob?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+class BatchJob:
+    """Batch Job entity
+
+    :param id: Fully qualified resource ID for the resource. Ex -
+     /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}.
+    :type id: str
+    :param name: The name of the resource.
+    :type name: str
+    :param type: The type of the resource. E.g. "Microsoft.Compute/virtualMachines" or
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py:22]: What does this actually mean? How do I invoke a vm batch job vs a storage batch job? I'm thinking if this is something we expose to customers or leave out for now.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job_property.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from typing import Dict
+
+# from azure.ai.ml._schema._deployment.batch.batch_job_property import BatchJobPropertySchema
+from azure.ai.ml._schema._deployment.batch.batch_deployment_settings import BatchRetrySettings
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job_property.py:8]: Is this a typo? Why BatchRetrySettings is imported from under _schema?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py (hunk 1 above)
Comment (TajiHarrisMicrosoft) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py:36]: Ah okay, I see what you mean now. Will make that fix

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py (hunk 2 above)
Comment (TajiHarrisMicrosoft) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py:22]: I got that definition from the BatchJobResource API contract, I'm not quite sure what that means. Maybe a better question for Fecund?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_endpoint_operations.py [hunk 4]
```diff
@@ -193,8 +204,9 @@ def invoke(
         endpoint_name: str,
         *,
         deployment_name: str = None,
-        inputs: Dict[str, Input] = None,
-        **kwargs,
+        input: Input = None,  # pylint: disable=redefined-builtin
```
Comment (TajiHarrisMicrosoft) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_endpoint_operations.py:207]: revert ot inputs, remove params_override

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py [hunk 5]
```diff
@@ -19,8 +22,8 @@
 from azure.ai.ml._utils._endpoint_utils import upload_dependencies, validate_scoring_script
 from azure.ai.ml._utils._http_utils import HttpPipeline
 from azure.ai.ml._utils._logger_utils import OpsLogger
-from azure.ai.ml._utils.utils import _get_mfe_base_url_from_discovery_service, modified_operation_client
-from azure.ai.ml.constants._common import AzureMLResourceType, LROConfigurations, ARM_ID_PREFIX
+from azure.ai.ml._utils.utils import _get_mfe_base_url_from_discovery_service, create_requests_pipeline_with_retry, download_text_from_url
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py:25]: Please remove unused imports

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py [hunk 6]
```diff
@@ -19,8 +22,8 @@
...
+from azure.ai.ml.constants._common import AzureMLResourceType, LROConfigurations, ARM_ID_PREFIX, API_URL_KEY
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py:26]: Remove unused import

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py [hunk 7]
```diff
@@ -215,22 +228,24 @@ def list_jobs(self, endpoint_name: str, *, name: str = None) -> ItemPaged[BatchJ
...
-                endpoint_name=endpoint_name,
-                deployment_name=name,
-                resource_group_name=self._resource_group_name,
-                workspace_name=self._workspace_name,
-                **self._init_kwargs,
-            )
 
-            # This is necessary as the paged result need to be resolved inside the context manager
-            return list(result)
+    def _get_workspace_location(self) -> str:
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py:245]: This is duplicated. Same function as the one declared in line 250. Please remove these changes.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/data_version.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+# ---------------------------------------------------------
+
+from typing import Dict, Union
+from azure.ai.ml._schema._deployment.batch.batch_job_property import OutputDataSchema
+from azure.ai.ml._schema.core.fields import ArmStr
+from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY
+
+
+class DataVersion:
+    """Batch Job Resource entity
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/data_version.py:12]: Please update the documentation properly. Seems that this was for Batch Job Resource not DataVersion

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/data_version.py [hunk 9]
```diff
@@ -0,0 +1,38 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from typing import Dict, Union
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/data_version.py:5]: Please remove unused import

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/compute_binding.py [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+from typing import Dict, Union
+
+from azure.ai.ml._schema._deployment.batch.compute_binding import ComputeBindingSchema
+from azure.ai.ml._schema.core.fields import ArmStr
+from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY
+
+
+class ComputeBinding:
+    """Batch Job Resource entity
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/compute_binding.py:13]: Please update documentation will associated ComputeBinding params and types.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job_property.py [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY
+from azure.ai.ml.entities._system_data import SystemData
+from azure.ai.ml._restclient.v2020_09_01_dataplanepreview.models import BatchJob as BatchJobData
+
+
+class BatchJobProperty:
+    """Batch Job Property entity
+
+    :param copmute: Compute configuration used to set instance count.
+    :type id: ComputeBinding
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job_property.py:19]:
```suggestion
    :type compute: ComputeBinding
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job_property.py [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+    :param dataset: Input dataset.
+    :type dataset: str
+    :param error_threshold: Error threshold, if the error count for the entire input goes above
+         this value, the batch inference will be aborted. Range is [-1, int.MaxValue] -1 value 
+         indicates, ignore all failures during batch inference.
+    :type error_threshold: int
+    :param input_data:Input data for the job.
+    :type input_date: Dict[str, JobInput]
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job_property.py:27]:
```suggestion
    :type input_data: Dict[str, JobInput]
```

Also, do we need to update line 46 to include the types too?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job_property.py [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+    :param mini_batch_size: Size of the mini-batch passed to each batch invocation.
+    :type mini_batch_size: int
+    :param name: name of batch job property
+    :type name: str
+    :param output_file_name: Output file name.
+    :type output_file_name: str
+    :param retry_settings: Retry Settings for the batch inference operation.
+    :type retry_settings: BatchRetrySettings
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job_property.py:35]:
```suggestion
    :type retry_settings: ~azure.ai.ml.entities._deployment.deployment_settings.BatchRetrySettings
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/system_data_schema.py [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+# pylint: disable=unused-argument,no-self-use
+
+import logging
+
+from marshmallow import fields
+
+from azure.ai.ml._schema.core.fields import StringTransformedEnum, UnionField, PathAwareSchema
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/batch/system_data_schema.py:11]: Please remove unused imports

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py [hunk 15]
```diff
@@ -2,36 +2,101 @@
...
+     "Microsoft.Storage/storageAccounts".
+    :type type: str
+    :param system_data: System data associated with resource provider.
+    :type system_data: SystemData
+    :param copmute: Compute configuration used to set instance count.
+    :type id: ComputeBinding
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py:29]:
```suggestion
    :type compute: ComputeBinding
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py [hunk 16]
```diff
@@ -2,36 +2,101 @@
...
+    :param dataset: Input dataset.
+    :type dataset: str
+    :param error_threshold: Error threshold, if the error count for the entire input goes above
+         this value, the batch inference will be aborted. Range is [-1, int.MaxValue] -1 value 
+         indicates, ignore all failures during batch inference.
+    :type error_threshold: int
+    :param input_data:Input data for the job.
+    :type input_date: Dict[str, JobInput]
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py:37]:
```suggestion
    :type input_data: Dict[str, JobInput]
```

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py [hunk 17]
```diff
@@ -2,36 +2,101 @@
...
+    def __init__(
+        self,
+        id: str = None,
+        name: str = None,
+        type: str = None,
+        system_data: SystemData = None,
+        compute: ComputeBinding = None,
+        dataset: str = None,
+        error_threshold: int = None,
+        input_data: Dict= None,
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_job.py:59]:
```suggestion
        input_data: Dict = None,
```

**Pull request closed without merging**